src/main/scala/com/microsoft/hyperspace/Hyperspace.scala

```diff
@@ -89,6 +89,20 @@ class Hyperspace(spark: SparkSession) {
     indexManager.refresh(indexName, mode)
   }
 
+  /**
+   * Optmizes index files by merging smaller files into larger files.
```

Comment:
`Optimize index by changing the underlying index data layout (i.e., compaction).`? (because we have different behaviors for full and quick). If  you are OK with this change, please change in other places where you use this comment.

Reply:
> (because we have different behaviors for full and quick).

thanks @imback82 , i didn't understand this part. The behavior is still the same right? merge smaller files into larger files? semantically that's the thing optimize does whether quick or full right?
Please correct me if I misunderstood.

Reply:
Yea, but it sounds kind of weird. Let's say we have two files per bucket with both 100MB. How are we defining "smaller" files? Also, if you have two bucket files (one 50MB, and the other 100MB), are we merging smaller (50MB) only? I think it's better to generalize what the optimize does, and be specific for each mode.

Reply:
I updated the comment as suggestd.